python/lsst/pipe/tasks/repair.py

```diff
@@ -234,22 +234,40 @@ def cosmicRay(self, exposure, keepCRs=None):
         except Exception:
             pass
 
-        mi = exposure.getMaskedImage()
-        bg = afwMath.makeStatistics(mi, afwMath.MEDIAN).getValue()
+        exposure0 = exposure            # initial value of exposure
+        binSize = self.config.cosmicray.background.binSize
+        nx, ny = exposure.getWidth()/binSize, exposure.getHeight()/binSize
+        if nx*ny <= 1:
```

Comment:
The way I see it, the current logic is much simpler for the `nx*ny<=1` case.  This is because the `estimateBackground` function returns an exposure and a `lsst.afw.math.mathLib.Background` object (computed based on the config values, for which the default is indeed set to MEDIAN here).  The variable `bg` fed into `findCosmicRays` is a double, so I would still need to compute that with `afwMath.makeStatistics(bg.getImageF(), afwMath.MEDIAN).getValue()` after having run `bg, exposure = measAlg.estimateBackground(exposure, self.config.cosmicray.background, subtract=False)` and note that I have set `subtract=False` in the above as we do not actually want a background subtracted copy of the image here (although I suppose we could do that as in the `nx*ny>1` case, but that seems like an extra operation that is not required for the majority of cases).

I can confirm that the MEDIAN sky level is the same for the case I gave above as it is in the current implementation.  

In all, I am inclined to leave the current logic.